extensions-core/kafka-indexing-service/src/test/java/org/apache/druid/indexing/kafka/supervisor/KafkaSupervisorIOConfigTest.java

```diff
@@ -119,6 +122,42 @@ public void testSerdeWithNonDefaults() throws Exception
     Assert.assertTrue("skipOffsetGaps", config.isSkipOffsetGaps());
   }
 
+  @Test
+  public void testSerdeForConsumerPropertiesWithPasswords() throws Exception
```

Comment:
I don't think this is actually testing what you want to be testing. It is basically testing that string entries in `consumerProperties` get serde'd correctly and (somewhat separately) that DefaultPasswordProvider works as intended.

What you might want to test is that a) `KafkaSupervisorIOConfig` gets deserialized properly when there is a `consumerProperties` entry that is _not_ a string (i.e. that is a map representing a `PasswordProvider` object) and b) that `KafkaIndexTask.addConsumerPropertiesFromConfig()` does what you expect it to do (i.e. ultimately converts this into a string that it can pass to the KafkaConsumer). Something along these lines:

```
    String jsonStr = "{\n"
                     + "  \"type\": \"kafka\",\n"
                     + "  \"topic\": \"my-topic\",\n"
                     + "  \"consumerProperties\": {\n"
                     + "    \"bootstrap.servers\": \"localhost:9092\",\n"
                     + "    \"ssl.truststore.password\": {\"type\": \"default\", \"password\": \"mytruststorepassword\"},\n"
                     + "    \"ssl.keystore.password\": {\"type\": \"default\", \"password\": \"mykeystorepassword\"},\n"
                     + "    \"ssl.key.password\": \"mykeypassword\"\n"
                     + "  }\n"
                     + "}";

    KafkaSupervisorIOConfig ioConfig = mapper.readValue(jsonStr, KafkaSupervisorIOConfig.class);

    Properties props = new Properties();
    KafkaIndexTask.addConsumerPropertiesFromConfig(props, mapper, ioConfig.getConsumerProperties());
```
(and then run your assertions). Bonus if you can test a PasswordProvider that is not the default one (I understand the EnvironmentVariablePasswordProvider is somewhat hard to test without some serious hacks).

Reply:
Fixed. I feel testing using `EnvironmentVariablePasswordProvider` here might make this method complex. We could think about creating a test helper utility that provides an instance of the `EnvironmentVariablePasswordProvider` and maybe replace the DefaultPasswordProvider usages in tests with this. This would be separate PR though. What do you think?

Reply:
Sounds reasonable to me. We're not really interested in testing the functionality of `EnvironmentVariablePasswordProvider` here, that should be done in `EnvironmentVariablePasswordProviderTest` (but it doesn't look like it actually tested it..)